README.md

```diff
@@ -11,6 +11,35 @@ This software is under active development and likely contains many bugs.
 Please open bugs on Github if you discover any issues with the software
 or documentation.
 
+Installation and Usage
+----------------------
+
+OnionGateway requires a working Go build environment. Once you have that configured you can build the `entry_proxy` and it's dependencies with the `go get` command. The binary will be built inside you `$GOPATH/bin` directory. You may need to add this directory your shell `$PATH` environment variable.
+
+```bash
+go get github.com/DonnchaC/oniongateway
+sudo entry_proxy
+```
+
+To improve performance, the server running the `entry_proxy` should have a Tor daemon which is running in `Tor2Web` mode. There are instructions for compiling Tor in this mode at https://github.com/globaleaks/Tor2web/wiki/Installation-Guide#build-tor-with-tor2web-mode-and-some-patches.
+
+`entry_proxy` uses the DNS system to resolve domain names to hidden service addresses. You should install a local caching DNS server to avoid making a DNS query for every client connection.
+
+Using a domain with OnionGateway
+--------------------------------
+
+To use a domain with OnionGateway you must configure your hidden service and point your domain at one or more `oniongateway` servers.
+
+Your hidden service should be configured to listen on port 443 with a valid CA-signed certificate for your public domain. For example it could present a valid cert for `myblog.com` from LetsEncrypt. You should configure your hidden service to also serve content directly to hidden service users over HTTP.
```

Comment:
IMHO, add an example of torrc:

```torrc
HiddenServiceDir /var/lib/tor/myblog/
HiddenServicePort 80 127.0.0.1:80
HiddenServicePort 443 127.0.0.1:443
```

and nginx.conf:

```nginx
server {
    listen 127.0.0.1:80;
    server_name .myblogaaaaaaaaaa.onion;

    include sites-available/myblog_com.inc;
}

server {
    listen 127.0.0.1:443;
    server_name .myblog.com;

    ssl on;
    ssl_certificate /path/to/myblog_com.crt;
    ssl_certificate_key /path/to/myblog_com.key;
    add_header Strict-Transport-Security "max-age=31536000; includeSubdomains";
    # TODO: HTTP Public Key Pinning (HPKP)

    include sites-available/myblog_com.inc;
}
```

And put actual config of the site to `sites-available/myblog_com.inc` not to repeat it.